airflow/cli/cli_parser.py

```diff
@@ -62,6 +63,13 @@
     # Do no re-raise the exception since we want the CLI to still function for
     # other commands.
 
+try:
+    auth_mgr = get_auth_manager()
```

Comment:
It looks like this helper gets the class from config and constructs an instance before returning. Could we update that to take an optional arg (or add another helper) to just return the class? For executors the `get_cli_commands` function is a static method and you just need the class imported to call it, no need to create an instance (i.e. run `__init__` and all it's side effects).

Also how expensive is `init_auth_manager` and the fab class itself to import? It's important to optimize anything on the CLI path to be very fast to import since it will slow down the CLI otherwise. You can see some examples of what I did in the executor space and the benchmarks I provided:
- Optimizing the [base executor class](https://github.com/apache/airflow/pull/30361)
- Optimizing [Celery](https://github.com/apache/airflow/pull/31001) and [Kubernetes](https://github.com/apache/airflow/pull/30727) executor classes.

Reply:
ah yes good point. We discussed it with Vincent, and he was in favor of a non-static method in case implementers want to use instance properties, but what you say about init time makes sense, I'll change it.

Reply:
I am good with that then

Reply:
@vandonr-amz Did you end up looking at the affect of the import time of the fab auth manager (second paragraph in my first comment)?

I think it's important to do at least some quick benchmarking to see how slow or fast it is. Before I optimized the executors they added seconds to the CLI runtime just because loading the module was slow.

Reply:
ok I will.

Reply:
taking a quick look, I put a timer at the top of `cli_parser.py` and another one just after the test on duplicate subcommands (at the end of top level code).
Without this code we're at ~30ms
With, it's taking ~90ms
So this costs ~60ms.

Reply:
I can bring this time down to 40-45ms by getting rid of the `from flask import url_for` in fab auth manager.